File: Microverse coding challenges/drawing-book.js

// Pseudo code | Strategy

// 1. Make an array of pages;
// 2. Count number of pages from the front to arrive to the page;
// 3. Count number of pages from back to arrive to the page;
// 4. If countFront > countBack => return countBack;
//  else return countFront;


function pageCount(n, p) {
  // 1. Make an array of pages;
  let pagesArr = [];
  for (let i = 0; i <= n; i++) {
    pagesArr.push(i);
  }

  let bookPages = [];
  for (let j = 1; j <= pagesArr.length; j++) {
    bookPages.push(pagesArr.splice(0, 2));
    if(pagesArr.length === 1) {
      bookPages.push(pagesArr.splice(0));
    }
  }
  
}

pageCount(6, 2);

File: bin/drawing-book.rb
# Pseudo code
# 1. Count front moves (f)
# 2. Count back moves (b) 
# 3. Devide by 2 and round to integer
# 4. Compare and return lowest one 

def pageCount(n, p)  
  f = p / 2
  n.even? ? b = (n + 1 - p) / 2 : b = (n - p) / 2

  if (f > b)
    b
  else
    f
  end
end

puts pageCount(16, 7) # => 0
